src/visualizations/config/generators/dhis/singleValue.js

```diff
@@ -34,6 +41,8 @@ const generateValueSVG = (value, formattedValue, subText, legendSet, y) => {
 
     const fillColor = legendSet
         ? getColorByValueFromLegendSet(legendSet, value)
+        : formattedValue === noData.text
```

Comment:
technically correct but maybe not the most readable to have nested ternaries. consider just keeping it simple with a couple of `if`'s for readability.

Reply:
Changed.